### ISSUE TITLE
Subscription Downgrades fail with the error “No Payment Method Provided”

### ISSUE THREAD

**Issue by meetshevene** [woocommerce/woocommerce-blocks#8790]
## Describe the bug

Downgrading a subscription that results in a $0 cost fails with the error message `No Payment Method Provided`.

## To reproduce

1. Enable subscription switching in **WooCommerce →  Settings →  Subscriptions**
2. Purchase a variable subscription product
3. Go to **My Account → Subscriptions** and choose to change the subscription frequency. 
4. Downgrade to a cheaper subscription. This should result in a $0 charge at checkout.
5. See error `No Payment Method Provided`. 

## Expected behavior

Subscription downgrades should be processed without the need for a payment method. 

### WordPress (please complete the following information):

-   WordPress version: 6.1.1
-   WooCommerce version: 7.5.0
-   WooCommerce Blocks version: 9.8.1
-   Site language: English
-   Any other plugins installed: WooCommerce Subscriptions, WooCommerce Stripe

### Desktop (please complete the following information):

-   OS: macOS
-   Browser: Chrome
-   Version 111.0.5563.64

## Additional context

We also see where the button text that should be `Switch Subscription` still says `Place Order`.

6042831-zen

**Comment by mikejolley** [comment]
For anyone investigating this, the Store API requires a payment method here: https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/StoreApi/Routes/V1/Checkout.php#L552-L566

This may be a subscriptions issue because this line should evaluate to false if the order total is 0, unless something is filtering it:

```
$requires_payment_method = $this->order->needs_payment();
```

Ref:

https://github.com/Automattic/woocommerce-subscriptions-core/blob/0263521548d129574145d015a171194a36b0c3a7/includes/class-wc-subscriptions-order.php#L39

https://github.com/Automattic/woocommerce-subscriptions-core/blob/0263521548d129574145d015a171194a36b0c3a7/includes/class-wc-subscriptions-order.php#L634-L671

**Comment by ralucaStan** [comment]
@mikejolley should we transfer this issue to the Subscriptions repo?

**Comment by mikejolley** [comment]
@ralucaStan yeah lets do that. If there is an issue our side they can pass it back; I'm not sure what the expectation is regarding the payment method code above.
